eelsdbconverter/__init__.py

```diff
@@ -49,60 +48,186 @@ def __init__(self):
-            data.get('download_link')
-        ]
-
-        reference = data.get('reference')
-        if reference:
-            if isinstance(reference, str):
-                archive.section_metadata.references.append(reference)
-            elif isinstance(reference, dict):
-                if 'freetext' in reference:
-                    archive.section_metadata.references.append(
-                        re.sub(r'\r+\n+', '; ', reference['freetext']))
-                else:
-                    archive.section_metadata.references.append('; '.join([
-                        str(value).strip() for value in reference.values()
-                    ]))
+            sample.elements = elements
 
-        section_data.repository_name = 'Electron Energy Loss Spectroscopy (EELS) database'
-        section_data.repository_url = 'https://eelsdb.eu'
-        section_data.entry_repository_url = data.get('permalink')
+        #Experiment
+        experiment = metadata.m_create(Experiment)
+        experiment.method_name = 'electron energy loss spectroscopy'
+        experiment.method_abbreviation = 'EELS'
+        experiment.experiment_publish_time = datetime.strptime(
+            file_data.get('published'), '%Y-%m-%d %H:%M:%S')
+        edges = file_data.get('edges')
+        if edges is not None:
+            if isinstance(edges, str):
+                edges = json.loads(edges)
+            experiment.edges = edges
+
+        #Instrument
+        instrument = metadata.m_create(Instrument)
+        instrument.source_label = file_data['microscope']
+        device_settings = instrument.m_create(DeviceSettings)
+        device_settings.device_name = file_data['microscope']
+        device_settings.max_energy = file_data['max_energy']
+        device_settings.min_energy = file_data['min_energy']
+        device_settings.guntype = file_data['guntype']
+        device_settings.beam_energy = file_data['beamenergy']
+        device_settings.resolution = file_data['resolution']
+        device_settings.step_size = file_data['stepSize']
+        device_settings.acquisition_mode = file_data['acquisition_mode']
+        device_settings.beam_current = file_data['beamcurrent']
+        device_settings.detector_type = file_data['detector']
+        device_settings.dark_current = file_data['darkcurrent']
+        
+
+        #Author Generated
+        author_generated = metadata.m_create(AuthorGenerated)
+        author_generated.permalink = file_data['permalink']
+        author_generated.author_name = file_data['author']['name']
+        author_generated.author_profile_url = file_data['author']['profile_url']
+        author_generated.author_profile_api_url = file_data['author']['profile_api_url']
+        author_generated.descriptionn = file_data['description']
+
+
+        
+
+        #Populating the data header; the information isn't in the metainfo file
+        #Data Header
+        
+        for i in range(len(df.columns)):
```

Comment:
This is unnecessary. We put the spectra definitions in the metainfo to avoid this. Not need for DataHeaders.